### PR TITLE
build: use also different K8s versions for E2E testing in GHA

### PR DESCRIPTION
**Pull request by tarilabs**
proposal for https://github.com/kubeflow/model-registry/issues/614

## Description
please note

> Note All include combinations are processed after exclude

from: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#excluding-matrix-configurations

## How Has This Been Tested?
see https://github.com/kubeflow/model-registry/pull/659#issuecomment-2546855431

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: .github/workflows/python-tests.yml [hunk 1]
```diff
@@ -66,12 +66,24 @@ jobs:
...
     strategy:
       fail-fast: false
       matrix:
-        python: ["3.12", "3.11", "3.10", "3.9"]
+        python: ["3.12"] # see below for versions 3.9-3.11
+        kubernetes-version: ["v1.27.11", "v1.28.7", "v1.29.2", "v1.30.6"]
+        exclude: # on main merges (not PRs), use also different K8s versions for E2E testing
+          - kubernetes-version: ${{ github.event_name != 'push' && 'v1.28.7' }}
+          - kubernetes-version: ${{ github.event_name != 'push' && 'v1.29.2' }}
+          - kubernetes-version: ${{ github.event_name != 'push' && 'v1.30.6' }}
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply .github/workflows/python-tests.yml:79]: It's a neat trick I discovered recently, but the "limitation" is that the order will always be 1. Exclusion, 2. Inclusion

So there is only so much can be done in 1 matrix

For more complex cases, the solution would be to produce the matrix in one job, and consume in another job--that's what also GHA docs advise in general 

For this case, this was compact enough imho

**Comment by Al-Pragliola** [comment]
nice work! /lgtm /approve

**Review by Al-Pragliola — APPROVED** [review]

**Review by andreyvelich — COMMENTED** [review]
Thank you for doing this @tarilabs!
/lgtm

**Review by lampajr — APPROVED** [review]
/lgtm
/approve

**Pull request merged**